dataplane/snappi/sai_dataplane.py

```diff
@@ -0,0 +1,334 @@
+        if return_time:
+            return operation_time
+
+    def start_capture(self):
+        """ Start a capture which was already configured.
+        """
+        capture_names = self.get_capture_port_names(self.configuration)
+        logging.info(f"Starting capture on ports {str(capture_names)}")
+        cs = self.api.capture_state()
+        cs.state = cs.START
+        self.check_warnings(self.api.set_capture_state(cs))
+
+    def stop_capture(self):
+        capture_names = self.get_capture_port_names(self.configuration)
+        logging.info(f"Stopping capture on ports {str(capture_names)}")
+        cs = self.api.capture_state()
+        cs.state = cs.STOP
+        self.check_warnings(self.api.set_capture_state(cs))
+
+    def get_all_captures(self):
+        """
+        Returns a dictionary where port name is the key and value is a list of
+        frames where each frame is represented as a list of bytes.
+        """
+        cap_dict = {}
+        for name in self.get_capture_port_names(self.configuration):
+            print("Fetching captures from port %s" % name)
+            request = self.api.capture_request()
+            request.port_name = name
+            pcap_bytes = self.api.get_capture(request)
+
+            cap_dict[name] = []
+            for ts, pkt in dpkt.pcap.Reader(pcap_bytes):
+                cap_dict[name].append(list(pkt))
+
+        return cap_dict
+
+    def start_traffic(self):
+        """ Start traffic flow(s) which are already configured.
+        """
+        ts = self.api.transmit_state()
+        ts.state = ts.START
+        logging.info('Start traffic')
+        res = self.api.set_transmit_state(ts)
+        assert self.api_results_ok(res), res
+
+    def stop_traffic(self):
+        """ Stop traffic flow(s) which are already configured.
+        """
+        ts = self.api.transmit_state()
+        ts.state = ts.STOP
+        logging.info('Stop traffic')
+        res = self.api.set_transmit_state(ts)
+        assert self.api_results_ok(res), res
+
+    # PAUSE api not supported by Ixia-C
+    def pause_traffic(self):
+        """ Pause traffic flow(s) which are already configured.
+        """
+        ts = self.api.transmit_state()
+        ts.state = ts.PAUSE
+        logging.info('Pause traffic')
+        res = self.api.set_transmit_state(ts)
+        assert self.api_results_ok(res), res
+
+    def is_traffic_stopped(self, flow_names=[]):
+        """
+        Returns true if traffic in stop state
+        """
+        fq = self.api.metrics_request()
+        fq.flow.flow_names = flow_names
+        metrics = self.api.get_metrics(fq).flow_metrics
+        return all([m.transmit == "stopped" for m in metrics])
+
+    def get_all_stats(self, print_output=True):
+        """
+        Returns all port and flow stats
+        """
+        print("Fetching all port stats ...")
+        request = self.api.metrics_request()
+        request.choice = request.PORT
+        request.port
```

Comment:
this can be replaced with `req.port.port_names = []`

similar suggestion for flow_metrics.

Reply:
fixed